src/py42/modules/securitydata.py

```diff
@@ -169,6 +178,82 @@ def search_file_events(self, query):
         file_event_client = self._microservices_client_factory.get_file_event_client()
         return file_event_client.search(query)
 
+    def _search_by_hash(self, hash, type):
+        query = FileEventQuery.all(type.eq(hash))
+        response = self.search_file_events(query)
+        return response[u"fileEvents"]
+
+    def _find_file_versions(self, md5_hash, sha256_hash):
+        file_event_client = self._microservices_client_factory.get_file_event_client()
+        pds_client = self._microservices_client_factory.get_preservation_data_service_client()
+        response = file_event_client.get_file_location_detail_by_sha256(sha256_hash)
+
+        if u"locations" not in response and not len(response[u"locations"]):
+            raise Py42ArchiveFileNotFoundError(u"PDS service can't find requested file.")
```

Comment:
This should probably be a different, new type of exception, this type of exception is intended for when the device and file path are known, which is not the case here.

A simple `Py42Error` would probably be fine here (no need for a new exception type, and the message should contain the hash that was being searched for.

Reply:
Done.